de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/resources/VirSatResourceSetTest.java

```diff
@@ -588,6 +591,131 @@ public void testLoadAllResources() {
+		assertTrue("Resource got deserialized from persistant storage", resSei2.isLoaded());
+		assertTrue("Resource got deserialized from persistant storage", resSei2_1.isLoaded());
+		assertTrue("Resource got deserialized from persistant storage", resSei2_1_1.isLoaded());
+		assertFalse("Non DVLM resource in resource set should not be loaded", nonDvlmResourceReload.isLoaded());
+		
+		// now the final test case, unloading all data and then see if they get properly reloaded
+		resSet.getResources().forEach((res) -> res.unload());
+		
+		resSet.loadAllResources();
+		
+		assertTrue("Resource got deserialized from persistant storage", resSei1.isLoaded());
+		assertTrue("Resource got deserialized from persistant storage", resSei2.isLoaded());
+		assertTrue("Resource got deserialized from persistant storage", resSei2_1.isLoaded());
+		assertTrue("Resource got deserialized from persistant storage", resSei2_1_1.isLoaded());
+		assertFalse("Non DVLM resource in resource set should not be loaded", nonDvlmResourceReload.isLoaded());
+	}
+	
+	@Test
+	public void testGetDvlmResources() throws IOException, CoreException {
+		VirSatResourceSet resSet = VirSatResourceSet.createUnmanagedResourceSet(testProject);
+		resSet.initializeModelsAndResourceSet();
+		
+		Repository repo = resSet.getRepository();
+		
+		StructuralElement se = StructuralFactory.eINSTANCE.createStructuralElement();
+		se.setIsApplicableForAll(true);
+		
+		//CHECKSTYLE:OFF
+		StructuralElementInstance sei1 = StructuralFactory.eINSTANCE.createStructuralElementInstance();
+		StructuralElementInstance sei2 = StructuralFactory.eINSTANCE.createStructuralElementInstance();
+		StructuralElementInstance sei2_1 = StructuralFactory.eINSTANCE.createStructuralElementInstance();
+		StructuralElementInstance sei2_1_1 = StructuralFactory.eINSTANCE.createStructuralElementInstance();
+		sei1.setType(se);
+		sei2.setType(se);
+		sei2_1.setType(se);
+		sei2_1_1.setType(se);
+		
+		Resource resSei1 = resSet.getAndAddStructuralElementInstanceResource(sei1);
+		Resource resSei2 = resSet.getAndAddStructuralElementInstanceResource(sei2);
+		Resource resSei2_1 = resSet.getAndAddStructuralElementInstanceResource(sei2_1);
+		Resource resSei2_1_1 = resSet.getAndAddStructuralElementInstanceResource(sei2_1_1);
+		//CHECKSTYLE:ON
+		
+		repo.getRootEntities().add(sei1);
+		repo.getRootEntities().add(sei2);
+
+		sei2.getChildren().add(sei2_1);
+		sei2_1.getChildren().add(sei2_1_1);
+		
+		resSet.saveAllResources(new NullProgressMonitor(), UserRegistry.getInstance());
+		
+		resSei1.unload();
+		resSei2.unload();
+		resSei2_1.unload();
+		resSei2_1_1.unload();
+		
+		URI uri = URI.createPlatformResourceURI("/testProject_VirSatResourceSetTest/test.ecore", true);
+		Resource newResource = new XMIResourceImpl(uri);
+		newResource.save(Collections.EMPTY_MAP);
+		Resource nonDvlmResourceReload = resSet.getResource(uri, true);
+		
+		assertTrue("Should contain DVLM resource", resSet.getDvlmResources().contains(resSei1));

```

Comment:
We can probably simplify this with `assertThat(message, resSet.getDvlmResources(), hasItems(...)`